drivers/flash/nrf_qspi_nor.c

```diff
@@ -440,12 +438,13 @@ static int qspi_nrfx_configure(const struct device *dev)
 		return -ENXIO;
 	}
 
+	struct qspi_nor_data *const driver_data = dev->driver_data;
```

Comment:
This variable should be named `dev_data` instead of `driver_data` (that's what is used in the call below and also in the event handler).
Also this `const` here does not seem to bring any value, and it's not consistent with other declarations of such variables, so I'd suggest to remove it.